src/collection-view.js

```diff
@@ -644,18 +693,15 @@ const CollectionView = Backbone.View.extend({
   },
 
   // Destroy the child views that this collection view is holding on to, if any
-  _destroyChildren({checkEmpty} = {}) {
-    this.triggerMethod('before:destroy:children', this);
-    const shouldCheckEmpty = checkEmpty !== false;
+  _destroyChildren(options) {
     const childViews = this.children.map(_.identity);
 
-    this.children.each(_.bind(this.removeChildView, this));
-
-    if (shouldCheckEmpty) {
```

Comment:
@paulfalgout I know why the test fail
since `_removeChildViews` replaced [_onCollectionRemove](https://github.com/marionettejs/backbone.marionette/pull/3105/files#diff-091acc5f607e2622bd6cabf705443fffL129),  `_removeChildViews` needs to run `this._checkEmpty();` when its done. `_destroyChildren` also uses `_removeChildViews` so I placed the `shouldCheckEmpty` logic in there also to avoid double execution. So I can't move it out anymore.